tests/test_models.py

```diff
@@ -0,0 +1,178 @@
+from pytest import raises
+
+
+class TestTransactionModel(object):
+    def test_validating_an_invalid_transaction(self, b):
+        from bigchaindb.models import Transaction
+
+        tx = Transaction.create([b.me], [b.me])
+        tx.operation = 'something invalid'
+
+        with raises(TypeError):
+            tx.validate(b)
+
+        tx.operation = 'CREATE'
+        tx.fulfillments = []
+        with raises(ValueError):
+            tx.validate(b)
+
+
+class TestBlockModel(object):
+    def test_block_initialization(self, monkeypatch):
+        from bigchaindb.models import Block
+
+        monkeypatch.setattr('time.time', lambda: 1)
+
+        block = Block()
+        assert block.transactions == []
+        assert block.voters == []
+        assert block.timestamp == '1'
+        assert block.node_pubkey is None
+        assert block.signature is None
+
+        with raises(TypeError):
+            Block('not a list or None')
+        with raises(TypeError):
+            Block(None, 'valid node_pubkey', 'valid timestamp',
+                  'not a list or None')
+
+    def test_block_serialization(self, b):
+        from bigchaindb_common.crypto import hash_data
+        from bigchaindb_common.util import gen_timestamp, serialize
+        from bigchaindb.models import Block, Transaction
+
+        transactions = [Transaction.create([b.me], [b.me])]
+        timestamp = gen_timestamp()
+        voters = ['Qaaa', 'Qbbb']
+        expected_block = {
+            'timestamp': timestamp,
+            'transactions': [tx.to_dict() for tx in transactions],
+            'node_pubkey': b.me,
+            'voters': voters,
+        }
+        expected = {
+            'id': hash_data(serialize(expected_block)),
+            'block': expected_block,
+            'signature': None,
+        }
+
+        block = Block(transactions, b.me, timestamp, voters)
+
+        assert block.to_dict() == expected
+
+    def test_block_invalid_serializaton(self):
+        from bigchaindb_common.exceptions import OperationError
+        from bigchaindb.models import Block
+
+        block = Block([])
+        with raises(OperationError):
+            block.to_dict()
+
+    def test_block_deserialization(self, b):
+        from bigchaindb_common.crypto import hash_data
+        from bigchaindb_common.util import gen_timestamp, serialize
+        from bigchaindb.models import Block, Transaction
+
+        transactions = [Transaction.create([b.me], [b.me])]
+        timestamp = gen_timestamp()
+        voters = ['Qaaa', 'Qbbb']
+        expected = Block(transactions, b.me, timestamp, voters)
```

Comment:
I think it is ok for now, as it is not used/needed in multiple places. We can tackle this later, if needed.